src/Microsoft.Health.Fhir.CosmosDb.Core/Features/Storage/StoreProcedures/StoredProcedureBase.cs

```diff
@@ -0,0 +1,41 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Azure.Cosmos;
+using Microsoft.Azure.Cosmos.Scripts;
+using Microsoft.Health.Core.Extensions;
+
+namespace Microsoft.Health.Fhir.CosmosDb.Core.Features.Storage.StoredProcedures
+{
+    public abstract class StoredProcedureBase : IStoredProcedure
+    {
+        private readonly IStoredProcedureMetadata _storedProcedureMetadata;
+
+        protected StoredProcedureBase(IStoredProcedureMetadata storedProcedure)
+        {
```

Comment:
We probably should add here a EnsureArg to validate if the stored procedure metadata is not null.

Reply:
done